### PR TITLE
PA-536: Track Progress internal outer API

### PR DESCRIPTION
**Pull request by benagain**
Non-public API endpoints to support Track Progress

### CODE REVIEW

**Review by Chazzysb — DISMISSED** [review]
Approved

File: src/TrackProgressInternal/SFA.DAS.TrackProgressInternal.Api.AcceptanceTests/ApiTests/UpdateKsbTest.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            .Select(x => x.RequestMessage)
+            .Where(x => x.Url.EndsWith("/populateksbs"));
+
+        request.First().Body.Should().BeValidJson<KsbHolder>()
+            .Which.Should().BeEquivalentTo(
+                new { Ksbs = course.Ksbs.ToArray() });
+    }
+
+    public record KsbHolder(IEnumerable<Course.Ksb> Ksbs);
+}
```
Comment (Pauljgraham) [inline src/TrackProgressInternal/SFA.DAS.TrackProgressInternal.Api.AcceptanceTests/ApiTests/UpdateKsbTest.cs:75]: I like this

File: src/TrackProgressInternal/SFA.DAS.TrackProgressInternal.Api.AcceptanceTests/ApiFixture.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
+﻿using AutoFixture;
+using SFA.DAS.TrackProgressInternal.Api.AcceptanceTests;
+
+namespace SFA.DAS.TrackProgressInternal.Tests;
+
+public class ApiFixture
+{
+    protected ApiFactory factory = null!;
+    protected Fixture fixture = null!;
+    protected HttpClient client = null!;
```
Comment (Pauljgraham) [inline src/TrackProgressInternal/SFA.DAS.TrackProgressInternal.Api.AcceptanceTests/ApiFixture.cs:10]: For protected types, resharper is  suggesting they should be capitalised ie Factory, Fixture... etc

**Review by Pauljgraham — COMMENTED** [review]
Looks very good, just a few tiny comments

File: src/TrackProgressInternal/SFA.DAS.TrackProgressInternal.Api/Controllers/CourseController.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+public class CourseController : ControllerBase
+{
+    private readonly IMediator mediator;
+
+    public CourseController(IMediator mediator) => this.mediator = mediator;
+
+    public record PopulateKsbsRequest(Guid[] KsbIds);
+
+    [HttpPost]
+    [Route("/courses/{standard}/populateksbs")]
```
[the remark this answers is not in the document]
Comment (benagain) [reply src/TrackProgressInternal/SFA.DAS.TrackProgressInternal.Api/Controllers/CourseController.cs:17]: Yep, I like that better

**Review by Pauljgraham — APPROVED** [review]

**Pull request merged**